test/v1/controllers/AmendLossClaimControllerSpec.scala

```diff
@@ -0,0 +1,182 @@
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package v1.controllers
+
+import play.api.libs.json.{JsValue, Json}
+import play.api.mvc.{AnyContentAsJson, Result}
+import uk.gov.hmrc.domain.Nino
+import uk.gov.hmrc.http.HeaderCarrier
+import v1.mocks.requestParsers.MockAmendLossClaimRequestDataParser
+import v1.mocks.services._
+import v1.models.des.LossClaimResponse
+import v1.models.domain.{AmendLossClaim, TypeOfClaim, TypeOfLoss}
+import v1.models.errors.{NinoFormatError, NotFoundError, RuleIncorrectOrEmptyBodyError, _}
+import v1.models.outcomes.DesResponse
+import v1.models.requestData.{AmendLossClaimRawData, AmendLossClaimRequest}
+
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.Future
+
+class AmendLossClaimControllerSpec
+  extends ControllerBaseSpec
+    with MockEnrolmentsAuthService
+    with MockMtdIdLookupService
+    with MockAmendLossClaimService
+    with MockAmendLossClaimRequestDataParser
+    with MockAuditService {
+
+  val correlationId = "a1e8057e-fbbc-47a8-a8b4-78d9f015c253"
+
+  val nino = "AA123456A"
+  val claimId = "AAZZ1234567890a"
+  val amendLossClaim = AmendLossClaim(TypeOfClaim.`carry-forward`)
+
+  val amendLossClaimResponse =
+    LossClaimResponse(Some("XKIS00000000988"), TypeOfLoss.`self-employment`, TypeOfClaim.`carry-forward`, "2019-20", "2018-07-13T12:13:48.763Z")
+
+  val lossClaimRequest: AmendLossClaimRequest = AmendLossClaimRequest(Nino(nino), claimId, amendLossClaim)
+
+  val responseBody: JsValue = Json.parse(
+    s"""
+      |{
+      |    "selfEmploymentId": "XKIS00000000988",
+      |    "typeOfLoss": "self-employment",
+      |    "taxYear": "2019-20",
+      |    "typeOfClaim": "carry-forward",
+      |    "lastModified": "2018-07-13T12:13:48.763Z"
+      |}
+    """.stripMargin)
+
+  val requestBody: JsValue = Json.parse(
+    s"""
+      |{
+      |  "typeOfClaim": "carry-forward"
+      |}
+    """.stripMargin)
+
+  trait Test {
+    val hc = HeaderCarrier()
+
+    val controller = new AmendLossClaimController(
+      authService = mockEnrolmentsAuthService,
+      lookupService = mockMtdIdLookupService,
+      amendLossClaimService = mockAmendLossClaimService,
+      amendLossClaimParser = mockAmendLossClaimRequestDataParser,
+      auditService = mockAuditService,
+      cc = cc
+    )
+
+    MockedMtdIdLookupService.lookup(nino).returns(Future.successful(Right("test-mtd-id")))
+    MockedEnrolmentsAuthService.authoriseUser()
+  }
+
+  "amend" should {
+    "return a successful response with header X-CorrelationId and body" when {
```

Comment:
Doesn't check the existence of a correlation ID - will need adding.  I'll do the other tests under a support ticket.

Reply:
I added for all the tests